sdk/dotnet/Pulumi/Deployment/Deployment_RegisterResourceOutputs.cs

```diff
@@ -25,8 +25,8 @@ void IDeploymentInternal.RegisterResourceOutputs(Resource resource, Output<IDict
 
             // The registration could very well still be taking place, so we will need to wait for its URN.
             // Additionally, the output properties might have come from other resources, so we must await those too.
-            var urn = await resource.Urn.GetValueAsync().ConfigureAwait(false);
-            var props = await outputs.GetValueAsync().ConfigureAwait(false);
+            var urn = await resource.Urn.GetValueAsync(whenUnknown: default!).ConfigureAwait(false);
```

Comment:
Leaving behavior as-before here, because I'm unsure. Should this not be running at all in preview? In that case we can opt for throwing an informative exception instead, saying this method is accidentally running in preview when it should not.